sigmastate/src/main/scala/sigmastate/serialization/ErgoTreeSerializer.scala

```diff
@@ -207,7 +207,12 @@ class ErgoTreeSerializer {
     val header = r.getByte()
     CheckHeaderSizeBit(header)
     val sizeOpt = if (ErgoTree.hasSize(header)) {
-      Some(r.getUIntExact)
+      val size = r.getUInt().toInt
+      // Note, when size < 0 as a result of Int overflow nothing happens here and later
+      // when deserialization proceeds normally as sizeOpt is not used on this pass.
+      // However, when ValidationException is thrown in deserializeErgoTree this negative
+      // size value will lead to undefined behavior
```

Comment:
What does undefined behavior means? Why can't it be defined by observing and reasoning on usages of the method ?

Reply:
added more comment in code